worldedit-core/src/main/java/com/fastasyncworldedit/core/queue/IBatchProcessor.java

```diff
@@ -23,7 +23,11 @@
      */
     IChunkSet processSet(IChunk chunk, IChunkGet get, IChunkSet set);
 
-    Future<IChunkSet> postProcessSet(IChunk chunk, IChunkGet get, IChunkSet set);
+    /**
+     * Post-process a chunk that has been edited. Set should NOT be modified here, changes will NOT be flushed to the world,
+     * but MAY be flushed to history. Defaults to nothing as most Processors will not use it.
+     */
+    default void postProcessSet(IChunk chunk, IChunkGet get, IChunkSet set) {}
```

Comment:
```suggestion
     * but MAY be flushed to history. Defaults to nothing as most Processors will not use it.
     *
     * @since TODO
     */
    default void postProcessSet(IChunk chunk, IChunkGet get, IChunkSet set) {}
```

Reply:
I suppose the method returning a Future ought to be kept actually, for API shizzle dizzle

Reply:
Likely, can deprecate it though.